src/gui/qgsmapcanvassnapper.cpp

```diff
@@ -169,10 +169,33 @@ int QgsMapCanvasSnapper::snapToBackgroundLayers( const QgsPoint& point, QList<Qg
   QSettings settings;
   QString snappingMode = QgsProject::instance()->readEntry( QStringLiteral( "Digitizing" ), QStringLiteral( "/SnappingMode" ), QStringLiteral( "current_layer" ), &snappingDefinedInProject );
   QString defaultSnapToleranceUnit = snappingDefinedInProject ? QgsProject::instance()->readEntry( QStringLiteral( "Digitizing" ), QStringLiteral( "/DefaultSnapToleranceUnit" ) ) : settings.value( QStringLiteral( "/qgis/digitizing/default_snapping_tolerance_unit" ), "0" ).toString();
-  QString defaultSnapType = snappingDefinedInProject ? QgsProject::instance()->readEntry( QStringLiteral( "Digitizing" ), QStringLiteral( "/DefaultSnapType" ) ) : settings.value( QStringLiteral( "/qgis/digitizing/default_snap_mode" ), "off" ).toString();
   QString defaultSnapTolerance = snappingDefinedInProject ? QString::number( QgsProject::instance()->readDoubleEntry( QStringLiteral( "Digitizing" ), QStringLiteral( "/DefaultSnapTolerance" ) ) ) : settings.value( QStringLiteral( "/qgis/digitizing/default_snapping_tolerance" ), "0" ).toString();
+  bool defaultSnapEnabled = settings.value( QStringLiteral( "/qgis/digitizing/default_snap_enabled" ), false ).toBool();
 
-  if ( !snappingDefinedInProject && defaultSnapType == QLatin1String( "off" ) )
+  QString defaultSnapType;
+  if ( snappingDefinedInProject )
+  {
+    defaultSnapType = QgsProject::instance()->readEntry( QStringLiteral( "Digitizing" ), QStringLiteral( "/DefaultSnapType" ) );
+  }
+  else
+  {
+    int dst;
+    dst = settings.value( QStringLiteral( "/qgis/digitizing/default_snap_mode" ), QgsSnappingConfig::Vertex ).toInt();
+    switch ( dst )
+    {
+      case 1:
+        defaultSnapType = QStringLiteral( "to segment" );
+        break;
+      case 2:
+        defaultSnapType = QStringLiteral( "to vertext and segment" );
+        break;
+      default:
+        defaultSnapType = QStringLiteral( "to vertext" );
+        break;
```

Comment:
Let's use QgsSnappingConfig::XYZ values in switch instead of numbers.

Also there are two typos above - "vertex" not "vertext"